fastapi_template/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/web/api/dummy/schema.py

```diff
@@ -13,9 +17,26 @@ class DummyModelDTO(BaseModel):
     It returned when accessing dummy models from the API.
     """
 
+    {%- if cookiecutter.db_info.name != "mongodb" %}
     id: int
+    {%- else %}
+    id: str
+    {%- endif %}
     name: str
 
+    {%- if cookiecutter.db_info.name == "mongodb" %}
+    @field_validator("id", mode="before")
```

Comment:
This thing won't work for old pydanticV1. 

You can check which pydantic version is currently being used by using `{%- if cookiecutter.pydanticv1 == "True" %} ...`

Reply:
Also, I'm wondering whether we need that validation in the end? Because as I think, pydantic will try to cast the value to string.

Reply:
> This thing won't work for old pydanticV1.

Good catch, I'll fix it.

>  I'm wondering whether we need that validation in the end?

We do need it, Pydantic models cannot store ObjectId unless we have
```
model_config = ConfigDict(extra="allow")
```

IMO allowing whatever third party class to be included in the model class seems a bit overkill, using a validator seems like a more reasonable way of doing it.

Reply:
Agree. Extra seems like a possible bug source in the future. Because they might change the type and handlers will no longer be able to return it in responses.